--- lib/python2.7/site-packages/gevent/ssl.py	2015-01-12 16:38:37.000000000 -0800
+++ ssl.py	2015-02-02 16:31:03.163586069 -0800
@@ -81,17 +81,15 @@
             self._sslobj = None
         else:
             # yes, create the SSL object
-            if ciphers is None:
-                self._sslobj = _ssl.sslwrap(self._sock, server_side,
-                                            keyfile, certfile,
-                                            cert_reqs, ssl_version, ca_certs)
-            else:
-                self._sslobj = _ssl.sslwrap(self._sock, server_side,
-                                            keyfile, certfile,
-                                            cert_reqs, ssl_version, ca_certs,
-                                            ciphers)
-            if do_handshake_on_connect:
-                self.do_handshake()
+            ctx = __ssl__.SSLContext(ssl_version)
+            if keyfile or certfile:
+                ctx.load_cert_chain(certfile, keyfile)
+            if ca_certs:
+                ctx.load_verify_locations(ca_certs)
+            if ciphers:
+                ctx.set_ciphers(ciphers)
+            self._sslobj = ctx._wrap_socket(self._sock, server_side=server_side)
+
         self.keyfile = keyfile
         self.certfile = certfile
         self.cert_reqs = cert_reqs
